pkg/apis/core/types.go

```diff
@@ -4142,11 +4142,48 @@ type NodeAddressType string
 
 // These are valid values of node address type
 const (
-	NodeHostName    NodeAddressType = "Hostname"
-	NodeExternalIP  NodeAddressType = "ExternalIP"
-	NodeInternalIP  NodeAddressType = "InternalIP"
-	NodeExternalDNS NodeAddressType = "ExternalDNS"
+	// NodeHostName identifies the node's canonical hostname
+	NodeHostName NodeAddressType = "Hostname"
+
+	// NodeInternalIP identifies an IP address which is assigned to one of the node's
+	// network interfaces. Every node should have at least one address of this type.
+	//
+	// An internal IP is normally expected to be reachable from every other node, but
+	// may or may not be visible to hosts outside the cluster. By default it is
+	// assumed that kube-apiserver can reach node internal IPs, though it is possible
+	// to configure clusters where this is not the case.
+	//
+	// In cloud environments, a NodeInternalIP address generally corresponds to an
+	// "internal" or "private" IP address which was automatically assigned to the
+	// node. In bare-metal clusters, NodeInternalIP is simply the standard address
+	// type, and does not necessarily imply that the IP is internal-only.
+	NodeInternalIP NodeAddressType = "InternalIP"
+
+	// NodeExternalIP identifies an IP address explicitly intended for use from
+	// outside the cluster; it is assumed to be more public or more persistent than a
+	// NodeInternalIP. (In public clouds, NodeExternalIP addresses are normally
+	// globally routable unicast IPs.)
+	//
+	// External IPs may be assigned directly to an interface on the node, like a
+	// NodeInternalIP, or alternatively, packets sent to the external IP may be NAT'ed
+	// to an internal node IP rather than being delivered directly (making the IP less
+	// efficient for node-to-node traffic than a NodeInternalIP).
+	//
+	// In cloud environments, a NodeExternalIP address generally corresponds to an
+	// "external", "public", or "virtual" IP address, and may have been manually
+	// assigned to the node by an administrator. In bare-metal clusters,
```

Comment:
I don't think  "bare-metal clusters" is accurate since this applies for cluster on cloud providers without the cloud provider integration and bare-metal clusters using a custom external cloud provider. I would rephrase this to say "In clusters without the cloud provider enabled,"

Reply:
updated